src/capi/impl/capi_sframe.cpp

```diff
@@ -229,20 +232,341 @@ EXPORT void tc_sframe_random_split(const tc_sframe* sf, double fraction,
+  ERROR_HANDLE_START();
+
+  return new_tc_sframe(sf->value.pack_columns(column_prefix, column_name, static_cast<turi::flex_type_enum>(type), value->value));
+
+  ERROR_HANDLE_END(error, NULL);
+}
+
+EXPORT tc_sframe* tc_sframe_split_datetime(const tc_sframe* sf, const char* expand_column, const char* column_prefix, const tc_flex_list* limit, bool tzone, tc_error** error){
+  ERROR_HANDLE_START();
+
+  std::vector<std::string> limit_transform;
+
+  for (const turi::flexible_type& i : limit->value){
+    if(i.get_type() != turi::flex_type_enum::STRING){
+      set_error(error, "limit is not of type str");
+      return NULL;
+    }
+
+    limit_transform.push_back(i.get<turi::flex_string>());
+  }
+
+  return new_tc_sframe(sf->value.split_datetime(expand_column, column_prefix, limit_transform, tzone));
+
+  ERROR_HANDLE_END(error, NULL);
+}
+
+EXPORT tc_sframe* tc_sframe_unpack(const tc_sframe* sf, const char* unpack_column, tc_error** error){
+  ERROR_HANDLE_START();
+
+  return new_tc_sframe(sf->value.unpack(unpack_column));
+
+  ERROR_HANDLE_END(error, NULL);
+}
+
+EXPORT tc_sframe* tc_sframe_unpack_detailed(const tc_sframe* sf, const char* unpack_column, const char* column_prefix, const tc_flex_enum_list* type, tc_flexible_type* value, const tc_flex_list* limit, tc_error** error){
+  ERROR_HANDLE_START();
+
+  std::vector<turi::flex_type_enum> type_transform;
+
+  for (const turi::flex_type_enum& i : type->value) {
+    type_transform.push_back(i);
+  }
+
+  return new_tc_sframe(sf->value.unpack(unpack_column, column_prefix, type_transform, value->value, limit->value));
+
+  ERROR_HANDLE_END(error, NULL);
+}
+
+EXPORT tc_sframe* tc_sframe_stack(const tc_sframe* sf, const char* column_name, tc_error** error){
+  ERROR_HANDLE_START();
+
+  return new_tc_sframe(sf->value.stack(column_name, column_name));
+
+  ERROR_HANDLE_END(error, NULL);
+}
+
+EXPORT tc_sframe* tc_sframe_stack_and_rename(const tc_sframe* sf, const char* column_name, const char* new_column_name, bool drop_na, tc_error** error){
+  ERROR_HANDLE_START();
+
+  return new_tc_sframe(sf->value.stack(column_name, new_column_name, drop_na));
+
+  ERROR_HANDLE_END(error, NULL);
+}
+
+EXPORT tc_sframe* tc_sframe_unstack(const tc_sframe* sf, const char* column, const char* new_column_name, tc_error** error){
+  ERROR_HANDLE_START();
+
+  return new_tc_sframe(sf->value.unstack(column, new_column_name));
+
+  ERROR_HANDLE_END(error, NULL);
+}
+
+EXPORT tc_sframe* tc_sframe_unstack_vector(
+    const tc_sframe* sf, const tc_flex_list* columns, const char* new_column_name, tc_error** error){
+  ERROR_HANDLE_START();
+
+  std::vector<std::string> columns_transform;
+
+  for (const turi::flexible_type& i : columns->value){
+    if(i.get_type() != turi::flex_type_enum::STRING){
+      set_error(error, "Columns is not of type str");
```

Comment:
Same as above -- 'element of specified columns is not of type str' 
Also use of `i`.